notes/stv/20210628-multi-user-deploy-01.txt

```diff
@@ -0,0 +1,153 @@
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2021, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+
+
+# Targets
+# -----------------------------------------------------
+  # Experiment with multiple users running jobs concurrently [Done]
+  # Experiment with SparkContexts [Done]
+  # Integrate changes into Ansible Scripts [Done]
+  # Test changed version with Example notebooks [Done]
+
+
+
+# -----------------------------------------------------
+# Create a container to work with.
+#[user@desktop]
+
+    source "${HOME:?}/aglais.env"
+
+    docker run \
+        --rm \
+        --tty \
+        --interactive \
+        --name ansibler2 \
+        --hostname ansibler \
+        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
+        --volume "${SSH_AUTH_SOCK}:/mnt/ssh_auth_sock:rw,z" \
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        atolmis/ansible-client:2020.12.02 \
+        bash
+
+
+# -----------------------------------------------------
+# Set the target cloud to delete.
+#[root@ansibler]
+
+    cloudname=gaia-test
+
+
+
+# -----------------------------------------------------
+# Delete everything.
+#[root@ansibler]
+
+    time \
+        /deployments/openstack/bin/delete-all.sh \
+            "${cloudname:?}"
+
+
+	> Done
+
+# -----------------------------------------------------
+# Create everything, using a standard config.
+#[root@ansibler]
+
+    time \
+        /deployments/hadoop-yarn/bin/create-all.sh \
+            "${cloudname:?}" \
+            'medium-04'
+
+
+	> Done
+
+
+
+# ----------------------------------------------------------------
+# Login via Firefox, in two separate windows, as two separate users
+#[user@desktop]
+
+    firefox --new-window "http://128.232.227.136:8080/" &
+
```

Comment:
Does this work ?
Not sure you can have two windows in the same browser logged in as different users.
When I tried it the second login changed the session cookie invalidating the first login.

I think we would either need to use a different browser for each user or use something like [container tabs](https://support.mozilla.org/en-US/kb/containers) in Firefox to manage the session cookies for each user.

Reply:
This works if you start it as an incognito window